src/view/components/AddPaymentEntry/index.tsx

```diff
@@ -0,0 +1,212 @@
+
+import { Farmer } from '../../../store/modules/farmer/types';
+import { PaymentEntry } from '../../../store/modules/payment/types';
+
+import { Route } from '../../navigation/routes';
+import { MapStateToProps, MapDispatchToProps, connect } from 'react-redux';
+import navActions from '../../../store/modules/nav/actions';
+import Composer from '../../hoc/PageComposer/index';
+import { State } from '../../../store/types';
+import paymentThunks from '../../../store/modules/payment/thunks';
+
+import Styles from './style';
+import { getActiveFarmer } from '../../../store/modules/farmer/selectors';
+
+/** AddPaymentEntry OwnPropsType */
+interface OwnPropsType {
+}
+
+/** AddPaymentEntry DispatchPropsType */
+interface DispatchPropsType {
+  navigate(route: Route): void;
+  goBack(): void;
+  createPaymentEntry(newEntry: PaymentEntry): Promise<string>;
+}
+
+/** AddPaymentEntry StorePropsType */
+interface StorePropsType {
+  farmer: Farmer;
+  activeTrader: any;
+  activeFarmer: any;
+}
+
+/** AddPaymentEntry PropsType */
+type PropsType = StorePropsType & DispatchPropsType & OwnPropsType;
+
+/** AddPaymentEntry OwnStateType */
+interface OwnStateType {
+  paymentAmount: number;
+  validAmount: boolean;
+}
+
+/** Button color */
+type ButtonColor = 'PRIMARY' | 'INFO';
+
+let radix: number = 10;
+
+/** AddPaymentEntry page */
+class AddPaymentEntry extends React.Component<PropsType, OwnStateType> {
+
+  private numbers = /^[0-9]+$/;
+  constructor(props: PropsType) {
+    super(props);
+    /** Init state */
+    this.state = {
+      paymentAmount: 0.0,
+      validAmount: false,
+    };
+  }
+  /** Get current datetime in specified format */
+  private getDatetime = (format: string) => moment().format(format);
+
+  /** Create page buttons */
+  private renderCancelButton = () => this.renderButton('Cancel', 'INFO', this.onCancelPress);
+  private renderSaveButton = () => this.renderButton('Save', 'PRIMARY', this.onSavePress);
+
+  /** Handle pressing cancel button */
+  private onCancelPress = () => this.props.goBack();
+
+  /** Handle pressing save button */
+  private onSavePress = () => {
+    // @TODO change time format to match core
+    const timeNow = moment().local().utc().toString();
+
+    /** Create a new Payment Entry */
+    let newEntry: PaymentEntry = {
+      type: 'payment',
+      datetime: timeNow,
+      toPersonUuid: this.props.activeFarmer,
+      fromPersonUuid: this.props.activeTrader,
+      amount: this.state.paymentAmount,
+      currency: 'UGX',
+    };
+    this.props.createPaymentEntry(newEntry);
+    this.props.navigate(Route.FARMER);
+  }
+
+  /** Return validity of required fields */
+  private allValid = () => (
+    this.state.validAmount
+  )
+
+  /** 
+   * Checks when the amount field has something in it 
+   * and verifies that it is a are real and positive number
+   */
+  private onChangeAmount = (newAmount: string) => {
+    const newAmountInt = parseInt(newAmount, radix);
```

Comment:
Now that I think of it, this should be a `float` not `int`. Please parse is as `float` and change all the `AddEntry` components to do the same.

You could use this as a reference
https://www.w3schools.com/Jsref/jsref_parsefloat.asp